sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroid.java

```diff
@@ -95,4 +102,36 @@ public static synchronized void init(
       throw new RuntimeException("Failed to initialize Sentry's SDK", e);
     }
   }
+
+  /**
+   * Deduplicate potentially duplicated Fragment and Timber integrations, which can be added
+   * automatically by our SDK as well as by the user. The user's ones win over ours.
+   *
+   * @param options SentryOptions to retrieve integrations from
+   */
+  private static void deduplicateIntegrations(final @NotNull SentryOptions options) {
+    final List<Integration> timberIntegrations = new ArrayList<>();
+    final List<Integration> fragmentIntegrations = new ArrayList<>();
+    for (final Integration integration : options.getIntegrations()) {
+      if (integration instanceof FragmentLifecycleIntegration) {
+        fragmentIntegrations.add(integration);
+      } else if (integration instanceof SentryTimberIntegration) {
+        timberIntegrations.add(integration);
+      }
```

Comment:
This is working because it is `compileOnly`, if you run this code without the dependencies, it will crash the App.
What if we filter out inside of the `SentryTimberIntegration` and `FragmentLifecycleIntegration` class?
The register method has access to the `options`

Reply:
oh, indeed... But how would I filter it out inside the integration that is going to be removed, actually? or you mean we'd just not register the integration if it's a duplicate one?

Reply:
Good point, didn't think about it, maybe just check if it really errors or not if both integrations are not in the classpath, it was a hunch that it'd crash, maybe for `instanceof` it does not.